### DISCUSSION TITLE
安装问题

### DISCUSSION THREAD

**Discussion by mianmianmi** [milvus-io/milvus#24036] [Q&A and General discussion]
可以有一个比较完整的安装步骤吗。我用Ubuntu+docker来安装Milvus，按照_https://milvus.io/docs/install_standalone-docker.md_步骤做了，三个容器也启动起来了。但是连不上去呀。
然后，我又去你们官网https://milvus.io/看到安装命令又是另外的
sudo apt install software-properties-common
sudo add-apt-repository ppa:milvusdb/milvus-standalone
sudo apt update
sudo apt install milvus
。。。
我建议能不能把安装步骤的逻辑整合好一些呀。对于我们这些第一次安装Milvus的来说，真的很不友好。比如Auto-gpt的安装指导就很明确，逻辑也很清晰。你们这个安装逻辑也太不清晰了吧。（也可能是我笨，所以，能麻烦安装步骤弄详细一点好吗。总之，我现在还是没有能连上Milvus）
![01](https://github.com/milvus-io/milvus/assets/61168978/6dd58ed0-255b-4bc8-b718-432bddbacaaf)
![02](https://github.com/milvus-io/milvus/assets/61168978/4b8189f5-0550-4df8-bc15-2c6a92dd25fa)
![03](https://github.com/milvus-io/milvus/assets/61168978/c13d1981-4e73-4393-93a6-1d4d08e53412)

**Comment by yhmo** [comment]
你看到的apt install的方式应该是2.1版本的页面： https://milvus.io/docs/v2.1.x/install_standalone-aptyum.md。
目前的2.2版本已经没有这种安装方式了。

2.2版本的安装最简单的方式是docker-compose： https://milvus.io/docs/install_standalone-docker.md，
其实就两行命令
```
wget https://github.com/milvus-io/milvus/releases/download/v2.2.8/milvus-standalone-docker-compose.yml -O docker-compose.yml
sudo docker-compose up -d
```


用Attu连接不上，我怀疑是Attu的版本和Milvus的版本不匹配。你看一下你的docker-compose.yml里写的是milvus哪个版本，使用的Attu镜像是哪个版本

**Reply by yhmo** [reply]
另外，如果你用2.1的apt install方式安装了老版的2.1，那就要应该把老版本的用apt remove卸载掉。

**Reply by mianmianmi** [reply]
好的，谢谢你。我今天就去试试.

**Reply by mianmianmi** [reply]
我觉得自己是傻逼，今天一次就装好了。

**Reply by gaoyuan5251** [reply]
请问这个安装2.2.x之前还需要单独安装 etcd and MinIO and Pulsar 么？

**Reply by yhmo** [reply]
不用单独安装 etcd and MinIO and Pulsar

**Reply by HarrisSK** [reply]
@mianmianmi 想问一下楼主我也是ubuntu系统通过docker-compose的方式安装milvus v2.2.14的，三个container也启动了，但就是status那里就是显示unhealthy，不知道是不是这个原因导致连不上去，楼主有遇到过吗？
<img width="1222" alt="image" src="https://github.com/milvus-io/milvus/assets/114902938/776db05d-f907-4a52-a4c3-56c19a5b7458">

**Reply by yhmo** [reply]
> @mianmianmi 想问一下楼主我也是ubuntu系统通过docker-compose的方式安装milvus v2.2.14的，三个container也启动了，但就是status那里就是显示unhealthy，不知道是不是这个原因导致连不上去，楼主有遇到过吗？ <img alt="image" width="1222" src="https://user-images.githubusercontent.com/114902938/265069695-776db05d-f907-4a52-a4c3-56c19a5b7458.png">


status unhealthy，这是docker层面的检查，可以用docker events查一下发生了什么

**Reply by HarrisSK** [reply]
![image](https://github.com/milvus-io/milvus/assets/114902938/6c7f3939-b1cb-4520-b166-4e8ccc23c843)
![image](https://github.com/milvus-io/milvus/assets/114902938/187d5584-3336-454a-8cca-a8c7d2f43263)
这是其中两个docker的情况，不知道为什么拒绝连接

**Reply by yhmo** [reply]
上面那个是minio的容器起不来，你在minio搜一下error看一下有没有相关issue。下面这个是etcd连不上，可能etcd容器也没起来

**Comment by guaimomotou123** [comment]
![image](https://github.com/milvus-io/milvus/assets/27824261/7f592f12-3b9f-48a6-865c-d2fd11de5582)
执行docker-compose up -d 这个命令，发现milvus-standalone这个容器总是启动不起来，上面截图是docker log，是什么原因呢

**Reply by yhmo** [reply]
截图上是milvus崩了之后，其中的两个goroutine被迫中断打出的栈，崩的时候应该会有大量这种gorutine的栈信息，但最初的原因是在这些信息的最上面，截图里没有截到最初的原因

**Reply by guaimomotou123** [reply]
<img width="1521" alt="image" src="https://github.com/milvus-io/milvus/assets/27824261/0896c2eb-8c3b-4a92-a60d-0f0c15b29cf1">
这个是最上面的信息，milvus的版本是v2.3.0

**Reply by yhmo** [reply]
还是RootCoord没有起来，再往上翻有没有别的ERROR或者WARN？

**Reply by guaimomotou123** [reply]
<img width="1525" alt="image" src="https://github.com/milvus-io/milvus/assets/27824261/bd43518e-1b92-4d9f-8893-f371642327c7">
这个是第一个WARN

**Reply by yhmo** [reply]
你把从启动到崩的日志打包发我邮箱吧，截图里没有看到rootcoord的情况。
正常的话，rootcoord会打出下面这些日志：
 [rootcoord/service.go:146] ["RootCoord init done ..."]
[2023/09/15 08:05:58.713 +00:00] [INFO] [rootcoord/service.go:288] ["RootCoord Core start ..."]

[2023/09/15 08:05:58.775 +00:00] [INFO] [rootcoord/root_coord.go:676] ["rootcoord startup successfully"]
[2023/09/15 08:05:58.775 +00:00] [DEBUG] [rootcoord/service.go:151] ["RootCoord start done ..."]
[2023/09/15 08:05:58.776 +00:00] [INFO] [components/root_coord.go:55] ["RootCoord successfully started"]

如果没有出现这些日志，那就是初始化过程中就崩了

**Reply by guaimomotou123** [reply]
大佬，已发送到zilliz.com你这个邮箱，docker全部启动日志已经在附件中，麻烦帮忙看一下，太感谢啦
